system/src/system_cloud_internal.cpp

```diff
@@ -844,7 +844,7 @@ bool system_cloud_active()
         if (SPARK_CLOUD_CONNECTED && ((now-lastCloudEvent))>SYSTEM_CLOUD_TIMEOUT)
         {
             WARN("Disconnecting cloud due to inactivity! %d, %d", now, lastCloudEvent);
-            cloud_disconnect(false); // TODO: Do we need to specify a reason of the disconnection here?
+            cloud_disconnect(HAL_PLATFORM_MAY_LEAK_SOCKETS ? false : true, false, CLOUD_DISCONNECT_REASON_ERROR);
```

Comment:
Are the true and false supposed to be other way round? Meaning, 
`HAL_PLATFORM_MAY_LEAK_SOCKETS ? true : false` because we now want to close socket on errors.

Reply:
The opposite. `HAL_PLATFORM_MAY_LEAK_SOCKETS` -> we cannot close sockets easily -> `cloud_disconnect(false)` -> do not close socket. This is for Electrons with SARA R4:

> Caveat: SARA R4 internal TCP/IP stack (used on Electrons) leaks sockets (opening/closing them too often will eventually prevent us from creating any more sockets). We leave the existing behavior on Electrons.

Reply:
If you have a better name for this feature flag in mind, I'm open to suggestions :smile:

Reply:
OK I understand. Yes, when I first read the `HAL_PLATFORM_MAY_LEAK_SOCKETS` it sounded like it may leak sockets, so fix it. 
I would rename it to something like `HAL_PLATFORM_ALLOWED_TO_LEAK_SOCKETS` but it's too long, so `HAL_PLATFORM_ALLOW_SOCKET_LEAK_IF_ANY` or how about just `HAL_PLATFORM_ALLOW_SOCKET_LEAK`

Reply:
I was trying to phrase the opposite that the platform (Electron with R4) itself leaks sockets. Not that we are leaking them, but the underlying network stack is buggy and we have to workaround it.